Automattic/wp-calypso
# Subscription management: Create a skeleton for individual site subscription page

Closes https://github.com/Automattic/wp-calypso/issues/76890

<p align="center">
  <table>
    <tr>
      <td>
        <img src="https://github.com/Automattic/wp-calypso/assets/2019970/1e12ccc3-1238-4460-b779-fdca5f2805d0" 
alt="Desktop view" width="400" />
        <p align="center"><em>Desktop view</em></p>
      </td>
      <td>
        <img src="https://github.com/Automattic/wp-calypso/assets/2019970/f78d6740-d015-4ed8-b0fb-fad8635afe6b" alt="Mobile view" width="400" />
        <p align="center"><em>Mobile view</em></p>
      </td>
    </tr>
  </table>
</p>

## Proposed Changes

* Introduce new individual site subscription page
  * Once connected to actual API's it will allow for previewing and editing site subscription settings
  * API's needed: 
    * Get site subscription data by siteId for the current user
    * Update site subscription settings
      *  we already have API's for updating each setting seperately, but we need one to update all of them at once (when "Save changes" button is pressed)

## Testing Instructions



1. Check out the PR.
2. Navigate to `client/server/pages/index.js` and temporarily add `return next();` to the line `833`:

![Markup on 2023-04-28 at 17:50:44](https://user-images.githubusercontent.com/25105483/235195033-4902bb9d-db57-43a4-b295-2d90feb6731f.png)

3. Build the app.
4. Log in to a test WordPress.com account that has some Sites subscriptions.
5. Navigate to http://calypso.localhost:3000/subscriptions/site/thisWillBeSiteID. The `thisWillBeSiteID` part of the URL does not really matter, it just hast to be some string for now. When we will finally integrate it with an API, a real site ID will have to be provided.
6. You should see the new page with mock data. 
7. Ensure that there is no regression on other pages, especially around popover settings.


_Credits to @ivan-ottinger for the above testing instructions 🙇_ 

## Pre-merge Checklist
